wapitiCore/attack/mod_log4shell.py

```diff
@@ -26,54 +28,94 @@ class ModuleLog4Shell(Attack):
+        for malicious_request, param_name, param_uuid in injected_get_and_post_requests:
+            try:
+                await self.crawler.async_send(malicious_request, follow_redirects=True)
+            except RequestError:
+                self.network_errors += 1
+                continue
+            await self._verify_param_vulnerability(malicious_request, param_uuid, param_name)
+
+    async def _verify_param_vulnerability(self, request: Request, param_uuid: uuid.UUID, param_name: str):
+        if not self._verify_dns(str(param_uuid)):
+            return
+        element_type = "query parameter" if request.method == "GET" else "body parameter"
+
+        await self.add_vuln_critical(
+            category=NAME,
+            request=request,
+            info=_("URL {0} seems vulnerable to Log4Shell attack by using the {1} {2}") \
+                .format(request.url, element_type, param_name),
+            parameter=f"{request.get_params if request.method == 'GET' else request.post_params}"
+        )
+
+        log_red("---")
+        log_red(
+            _("URL {0} seems vulnerable to Log4Shell attack by using the {1} {2}"),
+            request.url, element_type, param_name
+        )
+        log_red(request.http_repr())
+        log_red("---")
+
+    async def _verify_headers_vulnerability(
+        self,
+        modified_request:
+        Request, malicious_headers:
+        dict, headers_uuid_record: dict
+    ):
         for header, payload in malicious_headers.items():
             header_uuid = headers_uuid_record.get(header)
 
-            if await self._verify_dns(str(header_uuid)) is True:
+            if self._verify_dns(str(header_uuid)) is True:
                 await self.add_vuln_critical(
                     category=NAME,
                     request=modified_request,
-                    info=_("URL {0} seems vulnerable to Log4Shell attack by using the header {1}") \
-                        .format(modified_request.url, header),
+                    info=_("URL {0} seems vulnerable to Log4Shell attack by using the {1} {2}") \
+                        .format(modified_request.url, "header", header),
                     parameter=f"{header}: {payload}"
                 )
 
                 log_red("---")
                 log_red(
-                    _("URL {0} seems vulnerable to Log4Shell attack by using the header {1}"),
-                    modified_request.url, header
+                    _("URL {0} seems vulnerable to Log4Shell attack by using the {1} {2}"),
+                    modified_request.url, "header", header
                 )
                 log_red(modified_request.http_repr())
                 log_red("---")
 
-    async def _verify_dns(self, header_uuid: str) -> bool:
+    def _verify_dns(self, header_uuid: str) -> bool:
         resolver = dns.resolver.Resolver(configure=False)
         resolver.nameservers = [socket.gethostbyname(self.dns_endpoint)]
```

Comment:
`gethostbyname` on the `dns_endpoint` should be called only once.

Reply:
Fixed